library/std/src/collections/mod.rs

```diff
@@ -79,34 +79,45 @@
-//! experience worse performance.
-//!
-//! ## Sequences
-//!
-//! |                | get(i)                 | insert(i)               | remove(i)              | append    | split_off(i)           |
-//! |----------------|------------------------|-------------------------|------------------------|-----------|------------------------|
-//! | [`Vec`]        | *O*(1)                 | *O*(*n*-*i*)*           | *O*(*n*-*i*)           | *O*(*m*)* | *O*(*n*-*i*)           |
-//! | [`VecDeque`]   | *O*(1)                 | *O*(min(*i*, *n*-*i*))* | *O*(min(*i*, *n*-*i*)) | *O*(*m*)* | *O*(min(*i*, *n*-*i*)) |
-//! | [`LinkedList`] | *O*(min(*i*, *n*-*i*)) | *O*(min(*i*, *n*-*i*))  | *O*(min(*i*, *n*-*i*)) | *O*(1)    | *O*(min(*i*, *n*-*i*)) |
+//! Throughout the documentation, we will adhere to the following conventions for
+//! operation notation:
+//!
+//! * The collection's size is denoted by `n`.
+//! * If a second collection is involved, it's size is denoted by `m`.
+//! * Item indices are denoted by `i`.
+//! * Operations which have an *amortized* cost are suffixed with a `*`.
+//! * Operations with an *expected* cost are suffixed with a `~`.
+//!
+//! Calling operations that add to a collection will occasionally require a
+//! collection to be resized - an extra operation that takes *O*(*n*) time.
+//!
+//! *Amortized* costs are calculated to account for the time cost of such resize
+//! operations *over a sufficiently large series of operations*. An individual
+//! operation may be slower or faster due to the sporadic nature of collection
+//! resizing, however the average cost per operation will eventually equal the
+//! amortized cost.
+//!
+//! Rust's collections never automatically shrink, so removal operations aren't
+//! amortized.
+//!
+//! [`HashMap`] uses *expected* costs to denote that it is theoretically possible,
+//! though very unlikely, for [`HashMap`] to experience worse performance than
+//! the expected cost. This is due to the probablilistic nature of hashing -
+//! i.e. it is possible to generate a duplicate hash given some input key that
+//! requires extra computation to correct, but exceptionally unlikely.
+//!
+//! ## Cost of Sequence Operations
+//!
+//! |                | get(i)                 | insert(i)               | remove(i)              | append(Vec(m))    | split_off(i)           |
+//! |----------------|------------------------|-------------------------|------------------------|-------------------|------------------------|
+//! | [`Vec`]        | *O*(1)                 | *O*(*n*-*i*)*           | *O*(*n*-*i*)           | *O*(*m*)*         | *O*(*n*-*i*)           |
+//! | [`VecDeque`]   | *O*(1)                 | *O*(min(*i*, *n*-*i*))* | *O*(min(*i*, *n*-*i*)) | *O*(*m*)*         | *O*(min(*i*, *n*-*i*)) |
+//! | [`LinkedList`] | *O*(min(*i*, *n*-*i*)) | *O*(min(*i*, *n*-*i*))  | *O*(min(*i*, *n*-*i*)) | *O*(1)            | *O*(min(*i*, *n*-*i*)) |
```

Comment:
Do you think we should keep Sequences and Maps separate still? These two sets make it hard to fit in the last remaining oddball collection, BinaryHeap. It seems better to unify them into a single table.

You don't have to do that, I'm just asking because you're here, really. :^)